server/task/src/main/java/org/apache/james/task/MemoryTaskManager.java

```diff
@@ -38,36 +39,35 @@
 
     private static class DetailsUpdater implements TaskManagerWorker.Listener {
 
-        private final Consumer<TaskExecutionDetailsUpdater> updater;
+        private final Function<TaskId, Consumer<TaskExecutionDetailsUpdater>> updater;
```

Comment:
It's not an updater anymore.
And could you please name the type, Function doesn't describe the intent well enough in my opinion.